src/components/dashboard/add-qns/AddQuestionForm.tsx

```diff
@@ -0,0 +1,74 @@
+import { Button } from "@/components/ui/button";
+import { useState, SyntheticEvent, Dispatch, SetStateAction } from "react";
+import CustomInput from "@/components/form/CustomInput";
+import { Question, QuestionDifficulty } from "@/models/question.model";
+import DifficultySelect from "@/components/form/DifficultySelect";
+import CustomTextArea from "@/components/form/CustomTextArea";
+import "./AddQuestionForm.css";
+interface Props {
+  setIsAdding: Dispatch<SetStateAction<boolean>>;
+}
+function AddQuestionForm({ setIsAdding }: Props) {
+  const [title, setTitle] = useState<string>("");
+  const [description, setDescription] = useState<string>("");
+  const [complexity, setComplexity] = useState<QuestionDifficulty>(
+    QuestionDifficulty.Easy
+  );
+  const [link, setLink] = useState<string>("");
+  const [err, setError] = useState<string>("");
+
+  const onSubmit = (e: SyntheticEvent) => {
+    e.preventDefault();
+    if (title.length != 0 && description.length != 0 && link.length != 0) {
+      try {
+        const currStringify = localStorage.getItem("questions");
+        let curr: Question[] = [];
+        if (!currStringify) {
+          curr = [];
+        } else {
+          curr = JSON.parse(currStringify);
+        }
+
+        const newQuestion: Question = {
+          qId: curr.length ? curr.length + 1 : 1,
```

Comment:
This doesn't have to be removed yet, but just putting some thoughts here.

The frontend should not have to worry about setting the question id. Ideally, this is done in the backend, where your database will be the one assigning the new id.

LocalQuestionRepository should not assign ids when persisting to localStorage. Instead, your program should somehow call the backend to verify that the questions in the data table and the database are equal first (including the ids), then save this to localStorage / cache it somewhere.

But this is a bit complicated since it involves both Live and LocalQuestionRepository. Gotta think a bit, what do you think?